lib/Runtime/Library/PropertyString.cpp

```diff
@@ -48,6 +60,88 @@ namespace Js
         return this->hitRate > (int)CONFIG_FLAG(StringCacheMissThreshold);
     }
 
+    bool PropertyString::TryGetPropertyFromCache(
+        Var const instance,
+        RecyclableObject *const object,
+        Var *const propertyValue,
+        ScriptContext *const requestContext,
+        PropertyValueInfo *const propertyValueInfo)
+    {
+        if (ShouldUseCache())
+        {
+            PropertyValueInfo::SetCacheInfo(propertyValueInfo, this, GetLdElemInlineCache(), true);
+            if (CacheOperators::TryGetProperty<true, true, true, true, true, true, false, true, false>(
+                instance, false, object, this->propertyRecord->GetPropertyId(), propertyValue, requestContext, nullptr, propertyValueInfo))
+            {
+                LogCacheHit();
+#ifdef ENABLE_DEBUG_CONFIG_OPTIONS
+                if (PHASE_TRACE1(PropertyStringCachePhase))
+                {
+                    Output::Print(_u("PropertyCache: GetElem cache hit for '%s': type %p\n"), GetString(), object->GetType());
+                }
+#endif
+                return true;
+            }
+        }
+
+        LogCacheMiss();
+#ifdef ENABLE_DEBUG_CONFIG_OPTIONS
+        if (PHASE_TRACE1(PropertyStringCachePhase))
+        {
+            Output::Print(_u("PropertyCache: GetElem cache miss for '%s': type %p, index %d\n"),
+                GetString(),
+                object->GetType(),
+                GetLdElemInlineCache()->GetInlineCacheIndexForType(object->GetType()));
+            DumpCache(true);
+        }
+#endif
+        return false;
+    }
+
+    bool PropertyString::TrySetPropertyFromCache(
+        RecyclableObject *const object,
+        Var propertyValue,
+        ScriptContext *const requestContext,
+        const PropertyOperationFlags propertyOperationFlags,
+        PropertyValueInfo *const propertyValueInfo)
+    {
+        if (ShouldUseCache())
+        {
+            PropertyValueInfo::SetCacheInfo(propertyValueInfo, this, GetStElemInlineCache(), true);
+            if (CacheOperators::TrySetProperty<true, true, true, true, true, false, true, false>(
+                object,
```

Comment:
nit: can we be consistent with params on the same vs new line between this and the other function?